src/core/models/answersconfig.js

```diff
@@ -0,0 +1,214 @@
+/** @module AnswersConfig */
+
+import { LOCALE, PRODUCTION, QUERY_SOURCE, SANDBOX } from '../constants';
+import SearchConfig from '../models/searchconfig';
+import VerticalPagesConfig from '../models/verticalpagesconfig';
+import { AnswersConfigError } from '../errors/errors';
+
+const sandboxPrefix = `${SANDBOX}-`;
+const DEFAULTS = {
+  environment: PRODUCTION,
+  onReady: function () {},
+  useTemplates: true,
+  locale: LOCALE,
+  experienceVersion: 'PRODUCTION',
+  debug: false,
+  sessionTrackingEnabled: true,
+  onStateChange: function () {},
+  onVerticalSearch: function () {},
+  onUniversalSearch: function () {},
+  disableCssVariablesPonyfill: false,
+  querySource: QUERY_SOURCE
+};
+
+/**
+ * AnswersConfig is a model that is used that is used to turn a raw configuration into a data model that also storesVerticalPagesConfig}. It is used to initialze the Answers SDK
+ */
+class AnswersConfig {
```

Comment:
Let's break this file up into two classes: one for `AnswersConfig`, the other for `AnswersConfigBuilder`. It's a nicer separation of concerns: one class and one file for the config, another class and another file for the builder.